### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by Mateen-Ahmad5**
- Edit Entries
- Delete Entries
- Filter Entries

### CODE REVIEW

File: edit.js [hunk 1]
```diff
@@ -1,44 +1,125 @@
...
+let endDate = document.getElementById("end-date");
+let description = document.getElementById("description");
+
+let isEdit = false;
+
 function showExperienceForm() {
   document.getElementById("experience-form").style.display = "block";
 }
 document.getElementById("experience-form").addEventListener("submit", function (event) {
+  if (!company.value || !startDate.value || !endDate.value || !description.value) {
```
Comment (akbar-ashraf) [inline edit.js]: use validations here, for user better understanding

File: edit.js [hunk 2]
```diff
@@ -1,44 +1,125 @@
...
     event.preventDefault();
```
Comment (akbar-ashraf) [inline edit.js:17]: Remove duplication for preventdefault

File: edit.js [hunk 3]
```diff
@@ -1,44 +1,125 @@
...
+    return;
+  }
+
+  event.preventDefault();
+  
+    let experiences = JSON.parse(localStorage.getItem("experience")) || [];
+
+    let id = Date.now().toString(36) + Math.random().toString(36).substr(2, 9);
+
+    if (isEdit) {
```
Comment (akbar-ashraf) [inline edit.js:63]: is edit is being used as bolean at above and then used here as value, please recheck and fix

File: edit.js [hunk 4]
```diff
@@ -1,44 +1,125 @@
...
       });
-      document.getElementById("experience-form").reset();
+
       localStorage.setItem("experience", JSON.stringify(experiences));
-      document.getElementById("experience-form").style.display = "none";
-      loadExperiences();
     }
+
+    document.getElementById("experience-form").reset();
+    document.getElementById("experience-form").style.display = "none";
```
Comment (akbar-ashraf) [inline edit.js]: use the same approach as you have used to show like showExperienceForm.

File: edit.js [hunk 5]
```diff
@@ -1,44 +1,125 @@
...
+  isEdit = experience.id;
+  if (experience) {
+    company.value = experience.company;
+    description.value = experience.description;
+    startDate.value = experience.startDate;
+    endDate.value = experience.endDate;
+  }
+}
+
+function deleteExperience(id) {
```
Comment (akbar-ashraf) [inline edit.js:115]: use confirmation dialogue before deleting a record

**Review by akbar-ashraf — CHANGES_REQUESTED** [review]
Over all you have done a great job, however i have mentioned few improvements please make these changes and let me know

File: edit.js (hunk 1 above)
Comment (Mateen-Ahmad5) [reply edit.js]: update done

File: edit.js (hunk 2 above)
Comment (Mateen-Ahmad5) [reply edit.js:17]: fixed

File: edit.js (hunk 5 above)
Comment (Mateen-Ahmad5) [reply edit.js:115]: done

File: edit.js (hunk 4 above)
Comment (Mateen-Ahmad5) [reply edit.js]: Done

File: edit.js (hunk 3 above)
Comment (Mateen-Ahmad5) [reply edit.js:63]: done

**Review by akbar-ashraf — APPROVED** [review]
Good work!

**Pull request merged**